skada/datasets/_office_home.py

```diff
@@ -0,0 +1,338 @@
+        instead of trying to download the data from the source site.
+
+    shuffle : bool, default=False
+        If True the order of the dataset is shuffled.
+
+    random_state : int, RandomState instance or None, default=0
+        Determines random number generation for dataset shuffling. Pass an int
+        for reproducible output across multiple function calls.
+
+    Returns
+    -------
+    DomainAwareDataset : :class:`~skada.datasets.DomainAwareDataset`
+        Container carrying all dataset domains.
+    """
+    return _fetch_office_home_all(
+        _OFFICE_HOME_LOADER,
+        data_home=data_home,
+        download_if_missing=download_if_missing,
+        shuffle=shuffle,
+        random_state=random_state,
+    )
+
+
+def _fetch_office_home_all(
+    loader_spec: FileLoaderSpec,
+    data_home: Union[None, str, os.PathLike] = None,
+    download_if_missing: bool = True,
+    shuffle: bool = False,
+    random_state: Union[None, int, np.random.RandomState] = None,
+) -> DomainAwareDataset:
+    dataset = DomainAwareDataset()
+
+    data_home = get_data_home(data_home)
+    dataset_dir = os.path.join(data_home, loader_spec.dataset_dir)
+    if not os.path.exists(dataset_dir):
+        if not download_if_missing:
+            raise OSError("Data not found and `download_if_missing` is False")
+        os.makedirs(dataset_dir)
+        # juggling with `extract_root` is only required because SURF features
+        # were archived with root folder and DECAF without it
+        _download_office_home(
+            loader_spec.remote,
+            data_home,
+            data_home if loader_spec.extract_root else dataset_dir,
+        )
+
+    (X, y, domains) = _load_office_home_all(
+        dataset_dir,
+        shuffle=shuffle,
+        random_state=random_state,
+    )
+
+    for domain in OfficeHomeDomain:
+        domain_mask = np.char.lower(domains) == domain.value
+        X_domain = X[domain_mask]
+        y_domain = y[domain_mask]
+
+        dataset.add_domain(X_domain, y_domain, domain_name=domain.value)
+    return dataset
+
+
+def _fetch_office_home(
+    loader_spec: FileLoaderSpec,
+    domain: Union[OfficeHomeDomain, str],
+    data_home: Union[None, str, os.PathLike] = None,
+    download_if_missing: bool = True,
+    shuffle: bool = False,
+    random_state: Union[None, int, np.random.RandomState] = None,
+    return_X_y: bool = False,
+) -> Union[Bunch, Tuple[np.ndarray, np.ndarray]]:
+    if not isinstance(domain, OfficeHomeDomain):
+        domain = OfficeHomeDomain(domain.lower())
+    data_home = get_data_home(data_home)
+    dataset_dir = os.path.join(data_home, loader_spec.dataset_dir)
+    if not os.path.exists(dataset_dir):
+        if not download_if_missing:
+            raise OSError("Data not found and `download_if_missing` is False")
+        os.makedirs(dataset_dir)
+        # juggling with `extract_root` is only required because SURF features
```

Comment:
Based on this comment, seems like `extract_root` key is irrelevant for this dataset, right? 🤔